torch/utils/benchmark/utils/historic/CMakeLists.txt

```diff
@@ -0,0 +1,43 @@
+# Fallback for back-testing earlier versions of PyTorch.
+cmake_minimum_required(VERSION 3.18 FATAL_ERROR)
+
+project(timer_cpp_jit)
+set(CMAKE_CXX_STANDARD 14)
+
+message("name:            ${NAME}")
+message("torch_root:      ${TORCH_ROOT}")
+message("is_standalone:   ${IS_STANDALONE}")
+message("link_libtorch:   ${LINK_LIBTORCH}")
+message("valgrind_include ${VALGRIND_INCLUDE}")
```

Comment:
You weren't kidding about "not most beautiful thing in the world" :P

Reply:
You could make this marginally prettier by using `set()` to give these default values. Providing defaults can make it easier if people want to use this cmake to directly build code themselves or need to debug problems with the cmake.

Reply:
Would that help much? Bear in mind this is used for three different targets (depending on `${NAME}`), and most of the complexity is in the fiddly path magic that determines the values.